tasks/integration.yml

```diff
@@ -46,22 +46,91 @@
   loop: "{{ datadog_integration|dict2items }}"
   when: item.value.action == "remove" and ansible_facts.os_family == "Windows"
 
-# Install integrations
+# Check for existing integration installs / versions
 
-- name: Install pinned version of integrations (Unix)
-  command: "{{ datadog_agent_binary_path }} integration install {{ third_party }} {{ item.key }}=={{ item.value.version }}"
+- name: "Check for existing install / version of integrations (Unix)"
+  command:
+    argv:
+      - "{{ datadog_agent_binary_path }}"
+      - "integration"
+      - "show"
+      - "{{ integration.key }}"
+      - "-q"
   become: yes
   become_user: "{{ integration_command_user }}"
-  vars:
-    third_party: "{% if 'third_party' in item.value and item.value.third_party | bool %}--third-party{% endif %}"
+  check_mode: no
+  changed_when: >-
+    integration_results.rc != 0 or integration_results.stdout != integration.value.version
+
+  # Ignore command failure if integration is not already installed
+  failed_when: no
+  register: "integration_existing"
+  loop: "{{ datadog_integration | dict2items }}"
+  loop_control:
+    loop_var: 'integration'
+  when:
+    - "integration.value.action | lower == 'install'"
+    - "ansible_facts.os_family != 'Windows'"
+
+
+
+- name: "Check for existing install / version of integrations (Windows)"
+  win_command: '"{{ datadog_agent_binary_path }}" integration show {{ item.key }}'
+  become: yes
+  become_user: "{{ integration_command_user }}"
+  check_mode: no
+  changed_when: >-
+    integration_windows_existing.results.stderr is defined or ( integration_windows_existing.results.stdout is defined
+    and integration_windows_existing.results.stdout is not search ( item.value.version ) )
+
+  # Ignore command failure if integration is not already installed
+  failed_when: no
   loop: "{{ datadog_integration|dict2items }}"
-  when: item.value.action == "install" and ansible_facts.os_family != "Windows"
+  register: "integration_windows_existing"
+  when:
+    - "ansible_facts.os_family == 'Windows'"
+    - "item.value.action | lower == 'install'"
+
+- name: 'debug integration check'
+  debug:
+    var: 'integration_existing'
 
-- name: Install pinned version of integrations (Windows)
-  win_command: "\"{{ datadog_agent_binary_path }}\" integration install {{ third_party }} {{ item.key }}=={{ item.value.version }}"
+# Install integrations
+
+- name: "Install pinned version of integrations (Unix)"
+  command:
+    argv:
+      - "{{ datadog_agent_binary_path }}"
+      - "integration"
+      - "install"
+      - "{{ third_party }}"
```

Comment:
When `third_party` is empty, this makes the command fail (because it will be passed as an empty string argument). You need to use the one-line version.